lms/djangoapps/staticbook/tests.py

```diff
@@ -227,6 +227,23 @@ def test_static_url_map_static_asset_path(self):
         self.assertContains(response, 'file=/static/awesomesauce/{}'.format(
             PORTABLE_PDF_BOOK['chapters'][0]['url'].replace('/static/', '')))
 
+    def test_invalid_chapter_id(self):
+        """
+        Test that user is redirected when an invalid chapter id is provided.
+        """
+        pdf_textbooks=[PDF_BOOK, PORTABLE_PDF_BOOK]
+        self.make_course(pdf_textbooks=pdf_textbooks)
+        invalid_chapter = 99999
+        for book_index in xrange(len(pdf_textbooks)):
+            url = self.make_url('pdf_book', book_index=book_index, chapter=invalid_chapter)
+            response = self.client.get(url)
+            self.assertEqual(response.status_code,  302)
+            # verify that 0th  chapter of the same book is in the url
+            self.assertIn('{}/chapter/0' .format(book_index), response.url)
+            response = self.client.get(response.url)
+            self.assertEqual(response.status_code, 200)
+            self.assertContains(response, "Chapter 2 for PDF")
```

Comment:
No, when you access the 0th index chapter its means it should return the last chapter of this textbook. And here we have only two chapters, so it would return the second chapter.